### DISCUSSION TITLE
IncludeMany then中关联没有查到数据会产生一条id is null的sql

### DISCUSSION THREAD

**Discussion by misgw** [dotnetcore/FreeSql#1667] [Q&A]
``` c#
var parents = _freeSql
    .IncludeMany(p => p.Sons.Where(c => c.ParentId = p.Id), then => then.Include(p => p.Next))
    .ToList();
```
当关联Next, 并且Next中没有数据 就会产生一条这样的查询sql.
这种是否可以考虑直接设置为null?
``` sql
SELECT * FROM Next a WHERE (a.Id  IS  NULL)
```

**Comment by 2881099** [comment]
能提供一个可以重现的 console demo 吗，正常父级没数据， then 子级不会执行查询。

**Reply by misgw** [reply]
[TestInclude.zip](https://github.com/dotnetcore/FreeSql/files/13422301/TestInclude.zip)

**Reply by misgw** [reply]
不好意思,没注意看

**Reply by misgw** [reply]
测了一下, 发现好像不是IncludeMany的原因.是在Include没有时, 还要获取对应的数据时产生的查询

**Comment by misgw** [comment]
这种情况正常么

**Reply by 2881099** [reply]
提供一下demo，可重现你的情况，单纯文字描述不清楚表达什么
